### PR TITLE
sql-statement: improve consistency, fix small errors

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added or deleted? (Required)

* Improve consistency of the section `## MySQL compatibility`.
* Fix small errors

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions) (will manually cherry-pick to v3.1 as in https://github.com/pingcap/docs/pull/3663)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3585 (some of the changes are aligned in https://github.com/pingcap/docs-cn/pull/4375)
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: constraints.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ Query OK, 1 row affected (0.03 sec)
 
 ## 唯一约束
 
-在 TiDB 的乐观事务中，默认会对唯一约束进行[惰性检查](/transaction-overview.md#惰性检查)。通过在事务提交时再进行批量检查，TiDB 能够减少网络开销、提升性能。例如：
+根据不同的事务模式和 `tidb_constraint_check_in_place` 的值，TiDB 可能对唯一约束进行[惰性检查](/transaction-overview.md#惰性检查)。批量检查能够减少网络开销、提升性能。例如：
```
Comment (zz-jason) [inline constraints.md:62]: `默认` 描述了默认行为，换成 `可能` 后，缺少了一层意思

File: sql-statements/sql-statement-begin.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ Query OK, 0 rows affected (0.01 sec)
 
 ## MySQL 兼容性
 
-TiDB 在执行 `BEGIN` 语句时，可以带上 `PESSIMISTIC` 或者 `OPTIMISTIC` 选项表明此语句开启事务的类型。
+TiDB 支持 `BEGIN PESSMISTIC` 或 `BEGIN OPTIMISTIC` 的语法扩展，用户可以为某一个事务覆盖默认的事务模型。
```
Comment (lilin90) [inline sql-statements/sql-statement-begin.md:63]:
```suggestion
TiDB 支持 `BEGIN PESSIMISTIC` 或 `BEGIN OPTIMISTIC` 的语法扩展，用户可以为某一个事务覆盖默认的事务模型。
```

File: sql-statements/sql-statement-commit.md [hunk 3]
```diff
@@ -64,8 +64,8 @@ Query OK, 0 rows affected (0.01 sec)
 
 ## MySQL 兼容性
 
-* 在 MySQL 中，除了有多个 primary 的群组复制以外，`COMMIT` 语句通常不会导致错误。相比之下，TiDB 使用乐观并发控制，冲突可能导致 `COMMIT` 返回错误。
-* 默认情况下，`UNIQUE` 和 `PRIMARY KEY` 约束检查将延迟直至语句提交。可通过设置 `tidb_constraint_check_in_place=TRUE` 来改变该行为。
+* TiDB 3.0.8 及更新版本默认使用[悲观事务模型](/pessimistic-transaction.md)。在[乐观事务模型](/optimistic-transaction.md)下，需要考虑到修改的行已被另一个事务修改，导致`COMMIT` 语句可能执行失败的情况。
```
Comment (lilin90) [inline sql-statements/sql-statement-commit.md:67]:
```suggestion
* TiDB 3.0.8 及更新版本默认使用[悲观事务模型](/pessimistic-transaction.md)。在[乐观事务模型](/optimistic-transaction.md)下，需要考虑到修改的行已被另一个事务修改，导致 `COMMIT` 语句可能执行失败的情况。
```

File: sql-statements/sql-statement-create-sequence.md [hunk 4]
```diff
@@ -298,9 +298,9 @@ CREATE [TEMPORARY] SEQUENCE [IF NOT EXISTS] sequence_name
 
 ## MySQL 兼容性
 
-MySQL 暂无序列选项。TiDB 序列借鉴自 MariaDB。`SETVAL` 函数的步调是 TiDB 特有的，其他函数的步调与 MariaDB 保持一致。
+该语句是 TiDB 的扩展，实现序列借鉴自 MariaDB。
```
Comment (lilin90) [inline sql-statements/sql-statement-create-sequence.md:301]: How about the following version?
```suggestion
该语句是 TiDB 的扩展，序列的实现借鉴自 MariaDB。
```

File: sql-statements/sql-statement-drop-sequence.md [hunk 5]
```diff
@@ -50,7 +50,7 @@ Query OK, 0 rows affected (0.03 sec)
 
 ## MySQL 兼容性
 
-MySQL 暂无序列功能。
+该语句是 TiDB 的扩展，实现序列借鉴自 MariaDB。
```
Comment (lilin90) [inline sql-statements/sql-statement-drop-sequence.md:53]:
```suggestion
该语句是 TiDB 的扩展，序列的实现借鉴自 MariaDB。
```

File: sql-statements/sql-statement-show-create-sequence.md [hunk 6]
```diff
@@ -47,7 +47,7 @@ SHOW CREATE SEQUENCE seq;
 
 ## MySQL 兼容性
 
-MySQL 暂无序列选项。TiDB 序列部分借鉴自 MariaDB。`SETVAL` 函数的步调是 TiDB 特有的，其他函数的步调与 MariaDB 保持一致。
+该语句是 TiDB 的扩展，实现序列借鉴自 MariaDB。
```
Comment (lilin90) [inline sql-statements/sql-statement-show-create-sequence.md:50]:
```suggestion
该语句是 TiDB 的扩展，序列的实现借鉴自 MariaDB。
```

File: system-variables.md [hunk 7]
```diff
@@ -200,51 +200,44 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
...
 - 默认值：0
-- TiDB 支持乐观事务模型，即在执行写入时，假设不存在冲突。冲突检查是在最后 commit 提交时才去检查。这里的检查指 unique key 检查。
-- 这个变量用来控制是否每次写入一行时就执行一次唯一性检查。注意，开启该变量后，在大批量写入场景下，对性能会有影响。
+- 该变量仅适用于乐观事务模型。当这个变量设置为 0 时，唯一索引的重复值检查会被推迟到事务提交时才进行。这有助于提高性能，但对于某些应用，可能导致非预期的行为。详情见[约束](/constraints.md)。
 
-示例：
+    - 乐观事务模型下将 `tidb_constraint_check_in_place` 设置为 0：
 
-- 默认关闭 `tidb_constraint_check_in_place` 时的行为：
+        ```sql
```
Comment (lilin90) [inline system-variables.md:207]:
```suggestion
        {{< copyable "sql" >}}

        ```sql
```

File: system-variables.md [hunk 8]
```diff
@@ -200,51 +200,44 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
...
+        Query OK, 1 row affected
+        ```
 
-    ```sql
-    create table t (i int key);
-    insert into t values (1);
-    begin;
-    insert into t values (1);
-    ```
+        ```sql
```
Comment (lilin90) [inline system-variables.md:218]:
```suggestion
        {{< copyable "sql" >}}

        ```sql
```

**Review by lilin90 — COMMENTED** [review]
It seems that changes in `sql-statements/sql-statement-use.md` are missed. Please check. Thanks!

File: constraints.md (hunk 1 above)
Comment (lilin90) [reply constraints.md:62]: @zz-jason @ran-huang 感觉这里可以就保持原来的中文？意思还比较清楚🤔

Comment (ran-huang) [reply constraints.md]: Changes reverted. PTAL~

**Comment by lilin90** [comment]
I addressed two conflicts in https://github.com/pingcap/docs-cn/pull/4263/commits/20f722ae8ef5645cdf3bc9f4a35d43cf7d5bc040.

**Review by lilin90 — APPROVED** [review]
LGTM

File: constraints.md (hunk 1 above)
Comment (lilin90) [reply constraints.md]: LGTM

**Pull request merged**